### DISCUSSION TITLE
Github Desktop doesn't work with GPG signing commits

### DISCUSSION THREAD

**Discussion by williamhorning** [community/community#24082] [Apps, API and Webhooks]
<p>Whenever I try to make a commit in Github Desktop, it throws an error because it can’t open the GPG password screen.<br>
<img src="https://user-images.githubusercontent.com/68466727/181081851-572be6e8-5031-4976-ab33-97f894e67057.png" alt="image" data-base62-sha1="9MMRUNTEdLKZRhGIJ9FODwQ70Eo" width="386" height="186"></p>

**Comment by byrneh** [comment, marked as answer]
<p>hi <a class="mention" href="/wgyt">@wgyt</a> , welcome to the community</p>
<p>`GitHub Desktop does not support commit signing , this may have something to do with the error you are seeing.<br>
There is a ‘Note’ about this limitation at the following link <a href="https://docs.github.com/en/github/authenticating-to-github/signing-commits" rel="noopener nofollow ugc">signing-commits</a></p>

**Comment by williamhorning** [comment]
<aside class="quote no-group" data-username="byrneh" data-post="2" data-topic="171395">
<div class="title">
<div class="quote-controls"></div>
 byrneh:</div>
<blockquote>
<p>`GitHub Desktop does not support commit signing , this may have something to do with the error you are seeing.</p>
</blockquote>
</aside>
<p>Time to wait for them to add this feature. For now, using Visual Studio Code’s built in git support.</p>

**Comment by steveward** [comment]
<p><a class="mention" href="/wgyt">@wgyt</a> we’re in the process of adding GPG signing – you can follow along in <a href="https://github.com/desktop/desktop/issues/78">this issue</a> for updates about this feature.</p>
<p>GitHub Desktop should be able to sign your commits if you manually configure your GPG keys on <code>git</code>. You can read <a href="https://github.com/desktop/desktop/issues/78#issuecomment-460251896">this comment</a> for steps on how to do this. I hope that helps!</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/steveward">@steveward</a> , thanks for the update.<br>
Is anything happening about the documentation, which does state something different and without being psychic how is a normal user meant to discover this?</p>

**Comment by steveward** [comment]
<p><a class="mention" href="/byrneh">@byrneh</a> thanks for the feedback! This is undocumented because GPG signing has not been officially implemented into GitHub Desktop. It involves setting things up outside of the application, so it’s more of a workaround than an actual feature. Once we ship the feature to production we’ll have documentation to go along with it and the page on GPG signing will be updated accordingly.</p>
<p>The Gist with the workaround on how to set it up (<a href="https://gist.github.com/xavierfoucrier/c156027fcc6ae23bcee1204199f177da" class="inline-onebox">GPG signing with Git and Github Desktop · GitHub</a>) has some strong SEO as it’s the first result when you search for “GitHub Desktop GPG signing.” :grinning_face_with_smiling_eyes:</p>

**Comment by YeDemirkiran** [comment]
Still not solved in 2026
